src/transformers/models/resnet/modeling_resnet.py

```diff
@@ -416,3 +423,84 @@ def forward(
             return (loss,) + output if loss is not None else output
 
         return ImageClassifierOutputWithNoAttention(loss=loss, logits=logits, hidden_states=outputs.hidden_states)
+
+
+@add_start_docstrings(
+    """
+    ResNet backbone, to be used with frameworks like DETR and MaskFormer.
+    """,
+    RESNET_START_DOCSTRING,
+)
+class ResNetBackbone(ResNetPreTrainedModel):
```

Comment:
Would it be the case that we implement all backbone variations of ResNet in `modeling_resnet.py`? (I'd guess so?)

Reply:
We would just have a single backbone per model in the ilbrary. `ResNetBackbone` should cover all cases to use ResNet as a backbone, one can provide `config.out_features` to specify which feature maps to output

Reply:
OK. So in the case of DETR, when we have the frozenbatchnorm replacing batchnorm, we would load the resnet backbone, and apply the transformation in `modeling_detr.py` ?

Reply:
Yes, that's very specific to DETR so I'd to that there.